scripts/romanisim-make-stack

```diff
@@ -0,0 +1,267 @@
+            metadata['exposure']['ma_table_number'] = int(entry['ma_table_number'])
+            ma_table = parameters.ma_table[ metadata['exposure']['ma_table_number']]
+
+
+
+
+            # Loop over number of exposures
+            for entry_idx in range(int(entry['exposures'])):
+                # Set initial (possibly only) SCA
+                sca = args.sca if (args.sca > 0) else 1
+
+                # Date
+                date = Time(entry['date'], format='isot') + time_offset
+                metadata['exposure']['start_time'] = date
+
+                # If SCA value is within 1-18, only run once
+                # otherwise loop over all detectors
+                while (sca <= NUMBER_OF_DETECTORS):
+                    # Detector
+                    metadata['instrument']['detector'] = f'WFI{sca:02d}'
+
+                    # Create output file name
+                    output_file_name = f"roman_simulated_{entry['ra']}_{entry['dec']}_{date}_{entry['optical_element']}_{entry_idx + 1}_{sca}.asdf"
+
+                    if args.debug:
+                        print(f"output_file_name = {output_file_name}")
+
+                    # If making a script, write the line and proceed
+                    # otherwise create simulation file
+                    if args.make_script:
+                        line = f"romanisim-make-image {output_file_name}"
+
+                        for option, value in options_dct.items():
+                            if (type(value) != bool) and (option != 'sca'):
+                                line += f" --{option} {value}"
+                            elif (option == 'sca'):
+                                line += f" --{option} {sca}"
+                            elif (value == True):
+                                line += f" --{option}"
+
+                        line += f" --bandpass {entry['optical_element']}"
+                        line += f" --radec {entry['ra']} {entry['dec']}"
+                        line += f" --ma_table_number {entry['ma_table_number']}"
+                        line += f" --date "
+                        for s in date.strftime('%Y %m %d %H %M %S %f').split(' '):
+                            line += f"{int(s)} "
+
+                        # print line
+                        if args.debug:
+                            print(f'line = {line}')
+
+                        # add line to script
+                        spcript_file.write(line+'\n')
+
+                    else:
+                        # Obtain WCS catalog
+                        if args.catalog is None:
+                            twcs = wcs.get_wcs(metadata, usecrds=args.usecrds)
+                            rd_sca = twcs.toWorld(galsim.PositionD(
+                                galsim.roman.n_pix / 2, galsim.roman.n_pix / 2))
+                            cat = catalog.make_dummy_table_catalog(
+                                rd_sca, bandpasses=[entry['optical_element']], nobj=args.nobj)
```

Comment:
We don't want to make a new catalog for each entry / SCA.  That would mean that two images seeing the same sky would simulate different catalogs.  Probably the easiest thing to do is to just not support catalog == None.  Alternatively we need to generate a fake catalog outside both of these loops that covers the full footprint being simulated.  That seems a little messy, but we could imagine doing it.

Reply:
We could just require that any specified catalog spans the entire filed surveyed by the input pointings?

Reply:
Adjustment mentioned above.